### ISSUE TITLE
binary tables don't work in python 3

### ISSUE THREAD

**Issue by demorest** [demorest/sdmpy#19]
At least trying to unpack Pointing fails with python 3.  Haven't really looked into it yet.

**Comment by jmangum** [comment]
Hi Paul,

I am trying to extract total power atmospheric fluctuation information from the SysPower.bin files associated with Ka- and Q-band VLA measurements for an ngVLA study.  I think that I encountered this problem with python3 and binary tables:

```
cerussite:SkyStability jmangum$ ipython
Python 3.7.10 (default, Feb 26 2021, 10:16:00)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.22.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sdmpy

In [2]: import sdmpy.bintab

In [3]: s = sdmpy.SDM('/Users/jmangum/NRAO/ngVLA/TotalPower/SkyStability/19B-324.s
   ...: b38950032.eb38965904.59170.01636974537')
Traceback (most recent call last):

  File "/Users/jmangum/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-3-f521a7358d92>", line 1, in <module>
    s = sdmpy.SDM('/Users/jmangum/NRAO/ngVLA/TotalPower/SkyStability/19B-324.sb38950032.eb38965904.59170.01636974537')

  File "/Users/jmangum/anaconda3/lib/python3.7/site-packages/sdmpy-1.70.9-py3.7.egg/sdmpy/sdm.py", line 54, in __init__
    self._tables[tabname] = sdmtable(tabname, path, use_xsd=use_xsd)

  File "/Users/jmangum/anaconda3/lib/python3.7/site-packages/sdmpy-1.70.9-py3.7.egg/sdmpy/sdm.py", line 129, in sdmtable
    return SDMTable(name, path, *args, **kwargs)

  File "/Users/jmangum/anaconda3/lib/python3.7/site-packages/sdmpy-1.70.9-py3.7.egg/sdmpy/sdm.py", line 177, in __init__
    self._tree = objectify.parse(path+'/'+name+'.xml', parser)

  File "src/lxml/objectify.pyx", line 1842, in lxml.objectify.parse

  File "src/lxml/etree.pyx", line 3467, in lxml.etree.parse

  File "src/lxml/parser.pxi", line 1839, in lxml.etree._parseDocument

  File "src/lxml/parser.pxi", line 1865, in lxml.etree._parseDocumentFromURL

  File "src/lxml/parser.pxi", line 1769, in lxml.etree._parseDocFromFile

  File "src/lxml/parser.pxi", line 1163, in lxml.etree._BaseParser._parseDocFromFile

  File "src/lxml/parser.pxi", line 601, in lxml.etree._ParserContext._handleParseResultDoc

  File "src/lxml/parser.pxi", line 711, in lxml.etree._handleParseResult

  File "src/lxml/parser.pxi", line 640, in lxml.etree._raiseParseError

  File "<string>", line unknown
XMLSyntaxError: Element 'CalAntennaSolutionsTable': No matching global declaration available for the validation root.
```

Is the woraround to use python2.7?  Thanks.

-- Jeff

**Comment by jmangum** [comment]
Actually, it appears that the answer to my question is "no".  Installed for python 2.7 and get the same error:

```
In [3]: s = sdmpy.SDM('/Users/jmangum/NRAO/ngVLA/TotalPower/SkyStability/19B-324.s
   ...: b38950032.eb38965904.59170.01636974537')
  File "<string>", line unknown
XMLSyntaxError: Element 'CalAntennaSolutionsTable': No matching global declaration available for the validation root.
```

If it matters, this is rather recent (19B) Q-band data that I am trying to parse.  Thanks.

-- Jeff

**Comment by demorest** [comment]
Hi Jeff, for this issue try using `use_xsd=False` in `SDM()`.  I still need to update to the latest XML schema, this skips that check.

**Comment by jmangum** [comment]
Thanks Paul.  Works now (back to using python 3.7).  The unpack step took a minute or so to complete:
```
cerussite:SkyStability jmangum$ ipython
Python 3.7.10 (default, Feb 26 2021, 10:16:00)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.22.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sdmpy

In [2]: import sdmpy.bintab

In [3]: s = sdmpy.SDM('19B-324.sb38950032.eb38965904.59170.01636974537',use_xsd=Fa
   ...: lse)

In [4]: sp = sdmpy.bintab.unpacker(s['SysPower'])

In [5]: sp.unpack()

In [6]: sp.row
Out[6]:
rec.array([(b'Antenna_3', b'SpectralWindow_0', 0, 5112289462499999744, 1000000512, 2, [0.037849, 0.011714], [ 1.555887,  0.516333], [0.015625, 0.015625]),
           (b'Antenna_3', b'SpectralWindow_1', 0, 5112289462499999744, 1000000512, 2, [0.115592, 0.060679], [ 4.51995 ,  2.464903], [0.015625, 0.015625]),
           (b'Antenna_3', b'SpectralWindow_2', 0, 5112289462499999744, 1000000512, 2, [0.1645  , 0.086306], [ 6.55371 ,  3.341346], [0.015625, 0.015625]),
           ...,
           (b'Antenna_24', b'SpectralWindow_5', 0, 5112291344499999744,  999999488, 2, [0.534385, 0.426014], [15.705153, 11.586744], [0.015625, 0.015625]),
           (b'Antenna_24', b'SpectralWindow_6', 0, 5112291344499999744,  999999488, 2, [0.392068, 0.387982], [12.937733, 11.656777], [0.015625, 0.015625]),
           (b'Antenna_24', b'SpectralWindow_7', 0, 5112291344499999744,  999999488, 2, [0.287899, 0.305643], [10.736683, 10.860928], [0.015625, 0.015625])],
          dtype=[('antennaId', 'S32'), ('spectralWindowId', 'S32'), ('feedId', '<i4'), ('timeMid', '<i8'), ('interval', '<i8'), ('numReceptor', '<i4'), ('switchedPowerDifference', '<f4', (2,)), ('switchedPowerSum', '<f4', (2,)), ('requantizerGain', '<f4', (2,))])

In [7]:
```

**Comment by demorest** [comment]
Fixed in commit dbb5718 this was yet another bytes-versus-strings issue in python 3.
